src/pybind/mgr/prometheus/module.py

```diff
@@ -1817,6 +1813,38 @@ def setup_tls_using_cephadm(self, server_addr: str, server_port: int) -> None:
         self.set_uri(build_url(scheme='https', host=self.get_server_addr(),
                      port=server_port, path='/'))
 
+    def setup_tls_config_using_rook(self, server_addr: str, server_port: int) -> None:
+        cmd = {'prefix': 'orch certmgr generate-certificates',
```

Comment:
with the current code I see the code for `setup_tls_using_cephadm` is exaclty the same as for rook, so no need to create a new function as the logic is the same and we can have only one function `setup_https` or `setup_tls`